--- calamares.desktop
+++ calamares.desktop.src
@@ -3,14 +3,14 @@
 Version=1.0
 Name=Install System
 GenericName=System Installer
-Keywords=calamares;system;installer;
-TryExec=calamares
-Exec=sh -c "pkexec calamares"
-Comment=Calamares — System Installer
+Keywords=calamares;welcome;endeavouros;system;installer;
+TryExec=eos-welcome
+Exec=eos-welcome
+Comment=EndeavourOS Calamares — System Installer
 Icon=calamares
 Terminal=false
 StartupNotify=true
-Categories=Qt;System;
+Categories=System;
 X-AppStream-Ignore=true
 
 Name[ar]=تثبيت النظام 

