--- PartitionDialogHelpers.cpp	2021-09-06 14:44:52.000000000 +0200
+++ new-PartitionDialogHelpers.cpp	2021-09-07 19:37:28.261886573 +0200
@@ -23,15 +23,11 @@
 QStringList
 standardMountPoints()
 {
-    QStringList mountPoints { "/", "/home", "/opt", "/srv", "/usr", "/var" };
+    QStringList mountPoints { "/", "/boot", "/home", "/opt", "/srv", "/usr", "/var" };
     if ( PartUtils::isEfiSystem() )
     {
         mountPoints << Calamares::JobQueue::instance()->globalStorage()->value( "efiSystemPartition" ).toString();
     }
-    else
-    {
-        mountPoints << QStringLiteral( "/boot" );
-    }
     mountPoints.removeDuplicates();
     mountPoints.sort();
     return mountPoints;
