--- main.py	2021-10-23 23:19:15.406933947 +0200
+++ main-patched.py	2021-10-24 00:30:10.928136612 +0200
@@ -125,11 +125,11 @@
             # Replace HOOKS, MODULES and FILES lines with what we
             # have found via find_initcpio_features()
             if line.startswith("HOOKS"):
-                line = "HOOKS=\"{!s}\"".format(' '.join(hooks))
+                line = "HOOKS=({!s})".format(' '.join(hooks))
             elif line.startswith("MODULES"):
-                line = "MODULES=\"{!s}\"".format(' '.join(modules))
+                line = "MODULES=({!s})".format(' '.join(modules))
             elif line.startswith("FILES"):
-                line = "FILES=\"{!s}\"".format(' '.join(files))
+                line = "FILES=({!s})".format(' '.join(files))
             mkinitcpio_file.write(line + "\n")
 
 
@@ -145,7 +145,7 @@
     :return 3-tuple of lists
     """
     hooks = ["base", "udev", "autodetect", "modconf", "block", "keyboard", "keymap", "consolefont"]
-    modules = []
+    modules = ["vmd"]
     files = []
 
     swap_uuid = ""
